---
title: "MDS DGE analysis"
author: "Monica Ransom"
date: "8/23/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Pre-processing
1.    Adapters were trimmed using cutadapt v1.16
2.    Gene expression was quantified using salmon  v1.3.0
3.    TPMs were obtained for the genes using tximport 1.20.0

```{r import, message=FALSE}
library(dplyr)
library(ggplot2)
library(DESeq2)
library(tximport)
library(readr)
library(tximportData)
library(readxl)
library(knitr)
library(tidyverse)
library(pheatmap)
library(RColorBrewer)
library(viridis)
library(ggrepel)
library(EnhancedVolcano)

```
## R Markdown



When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. 

This section of code will generate a summary table of the samples sequenced and their sequencing and alignment metrics
```{r summary, echo=FALSE}
MDS_metric <- read_excel("~/Desktop/Jordan files/Results/MDS stats.xlsx")

starting_metrics<-filter(MDS_metric,grepl('HTB314|MDS268|MDS280', patient))
kable(starting_metrics, caption="Summary of Data Metrics")
```

This section of code will import and format the data for DeSeq2

```{r data, message=FALSE}
#to generate a vector of names and file locations
files<-file.path("~/Desktop/Jordan files/Counts/salmon/salmon/", list.files("~/Desktop/Jordan files/Counts/salmon/salmon/"), "quant.sf")
names(files)<-list.files("~/Desktop/Jordan files/Counts/salmon/salmon")

#to call in a gene_map this was derived by pulling it out of the fasta file with a grep command for ENST and ENSG and pasting them together
gene_map=read_csv("~/Desktop/Jordan files/Counts/salmon/gene_map.csv", col_names=c('enstid', 'ensgid'))

# import transcript level counts
txi.salmon.t<-tximport(files, type="salmon", txOut=TRUE)
txi.salmon.g<-tximport(files=files, type="salmon", tx2gene= gene_map, ignoreTxVersion = TRUE, countsFromAbundance = 'lengthScaledTPM' )
### this code works but if I remove the ignoreTxVersion I get an error, this may have to do with how I am generating my tx2gene file

# Extract counts only
counts <- txi.salmon.g$counts %>%
  as.data.frame()
#Extract TPM
tpms <- data.frame(txi.salmon.g$abundance)

##for clients the counts and tpm files should be written out

```

This chunk of code sets an expression threshold for TMP where all samples have at least 5 counts
```{r expressed genes}

expressed_genes<-tpms %>% filter_all(all_vars(. > 5))
```

This chunk of code generates a heatmap using the spearman method as well as a correlation heatmap
```{r All samples Heatmap, echo=FALSE, fig.width=7.5, fig.height=7, fig.align='center'}
## what we see here shows that patients cluster more closely than the cell types and this will need to be accounted for in the DGE
pheatmap(expressed_genes, scale="row")
cols.cor<-cor(expressed_genes, method="spearman")

pheatmap(cols.cor, method="spearman")
```
## PCA plot



```{r pca plot, message=FALSE}
exp.pca <- prcomp(t(log2(expressed_genes)))
exp.pca.summary <- summary(exp.pca)$importance
pc1var = round(exp.pca.summary[3,1] * 100, 1)
pc2var = round(exp.pca.summary[3,2] * 100 - pc1var, 1)
exp.pca.pc <- data.frame(exp.pca$x, sample = colnames(expressed_genes))

```

### PC1 vs PC2
```{r, All samples PCA 1 vs 2, echo = FALSE, fig.width= 6, fig.height=4.5, fig.align='center'}
metadata<-read_table2("~/Desktop/Jordan files/Counts/sample.txt")

ggplot(exp.pca.pc, aes(x = PC1, y = PC2, color=metadata$cellType, shape=metadata$patient)) +
  geom_point(size = 5)+
  xlab(paste('PC1,', pc1var, '% explained var.')) +
  ylab(paste('PC2,', pc2var, '% explained var.')) +
  geom_text_repel(aes(label=metadata$SampleName), size=3, max.overlaps = Inf)+
  ggtitle("Expressed Genes in All Samples > 5 TPM") 
```
## 3. Run Differential Expression testing using DESeq2 and Calculate Gene Set Enrichment
## Compare 123pos vs 123neg, 123neg vs bulk, and 123pos vs bulk 
#### sig = padj <0.01 and abs(l2fc) >0.5
#### 

```{r DGE}
coldata<-dplyr::select(metadata, -FileName) 
coldata<-column_to_rownames(coldata, 'SampleName')
## need to confirm that all names are in the same order
all(rownames(coldata) %in% colnames(txi.salmon.g$counts))
all(rownames(coldata) == colnames(txi.salmon.g$counts))
coldata<-coldata[colnames(txi.salmon.g$counts),]
all(rownames(coldata) == colnames(txi.salmon.g$counts))

dds <- DESeqDataSetFromTximport(txi.salmon.g,
                              colData = coldata,
                              design = ~ patient + cellType)
### ****Do I want to add filter here prior to running (Krysta adds genes have 10 counts)
dds <- DESeq(dds)
res <- results( dds )


### need to subset to do pairwise comparison unclear if this keeps the patient design aspect


res_123posvs123neg<-results(dds, contrast=c("cellType", "123pos", "123neg"))
res_123posvsBulk<-results(dds, contrast=c("cellType", "123pos", "bulk"))
res_123negvsBulk<-results(dds, contrast=c("cellType", "123neg", "bulk"))

## look at the numbers of genes meeting threshold the log fold change call is not changing things
sum( res_123posvs123neg$pvalue < 0.01 & abs(res_123posvs123neg$log2FoldChange) >= 0.5, na.rm=TRUE )
sum( res_123posvsBulk$pvalue < 0.01 & abs(res_123posvsBulk$log2FoldChange) >= 0.5, na.rm=TRUE )
sum(res_123negvsBulk$pvalue < 0.01 & abs(res_123negvsBulk$log2FoldChange) >= 0.5, na.rm=TRUE )
```
### Volcano Plot
```{r, echo=FALSE, fig.width= 10, fig.height=6}


df<-data.frame(res_123posvs123neg)
ggplot(df, aes(x=log2FoldChange, y=-log10(padj))) + 
  geom_point() +
  theme_light() +
  scale_colour_manual(values = colors) 
 
```

###MA plots
```{r, echo=FALSE, fig.width= 6, fig.height=4.5}

plotMA(res_123posvs123neg, 
       ylim=c(-3,3))
plotMA(res_123posvsBulk, 
       ylim=c(-3,3))
plotMA(res_123negvsBulk, 
       ylim=c(-3,3))

```

```{r pca plot 2, message=FALSE}
pca_data<-vst(dds, blind=T)
ntop=500
rv <- rowVars(assay(pca_data))
select <- order(rv, decreasing=TRUE)[seq_len(min(ntop, length(rv)))]
pca <- prcomp(t(assay(pca_data)[select,]))
percentVar <- data.frame(percentVar=pca$sdev^2/sum(pca$sdev^2))%>%
  mutate(pc=1:n())%>%dplyr::select(pc, percentVar)
pca_df<-pca$x%>%data.frame()%>%mutate(type=metadata$cellType)

alt_col_values=c("#88CCEE", "#CC6677", "#DDCC77")
```

### PC1 vs PC2
```{r, pca 2 plot, echo = FALSE, fig.width= 6, fig.height=4.5}
ggplot(pca_df, aes(PC1, PC2, color=type))+geom_point(size=3)+
  ##xlab(paste0("PC1: ",round(percentVar2$percentVar2[1] * 100), "% variance")) + 
  ##ylab(paste0("PC2: ", round(percentVar2$percentVar2[2] * 100), "% variance"))+
  scale_color_manual(values=alt_col_values)+labs(color="")+theme_bw()
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
