---
title: "MDS_DGE_HHT"
author: "Monica Ransom"
date: "9/7/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
This analysis is to look at the MDS samples that were treated with the drug HTT homoharringtonine which is a translation inhibitor.  There were only 2 samples for each treatment in this arm.  


# Pre-processing
1.    Adapters were trimmed using cutadapt v1.16
2.    Gene expression was quantified using salmon  v1.3.0
3.    TPMs were obtained for the genes using tximport 1.20.0

```{r import, message=FALSE, warning=FALSE}
library(dplyr)
library(ggplot2)
library(DESeq2)
library(tximport)
library(readr)
library(tximportData)
library(readxl)
library(knitr)
library(tidyverse)
library(pheatmap)
library(RColorBrewer)
library(viridis)
library(ggrepel)
library(EnhancedVolcano)
library(fgsea)
library(limma)
library(VennDiagram)
library(UpSetR)
library(wesanderson)
library(kableExtra)
library(reshape)
```


```{r, echo=FALSE, include=FALSE}
## Define Color Scheme and plot them
plasma <- viridis::plasma(n = 10)
viridis <- viridis::viridis(n = 10)
cividis <- viridis::cividis(n=10)
magma <- viridis::magma(n=10)
```

## Summary of Data Metrics

```{r summary, echo=FALSE}
MDS_metric <- read_excel("~/Desktop/Jordan files/Results/MDS stats.xlsx")

starting_metrics<-filter(MDS_metric,grepl('HTB61|HTB336', patient))

starting_metrics<-subset(starting_metrics, Sample!="CD123-" & Sample!="Bulk")
kable(starting_metrics, caption="Summary of Data Metrics")


```
This table shows that there are relatively high duplication levels in these samples.  This likely indicates that the input was low or too many PCR cycles were performed.  

## Summary of Data prior to analysis

```{r data, message=FALSE, echo=FALSE}
#to generate a vector of names and file locations
files<-file.path("~/Desktop/Jordan files/counts.rRNA/HHT/", list.files("~/Desktop/Jordan files/counts.rRNA/HHT/"), "quant.sf")
names(files)<-list.files("~/Desktop/Jordan files/counts.rRNA/HHT")

mart <- biomaRt::useMart("ENSEMBL_MART_ENSEMBL", dataset = "hsapiens_gene_ensembl", host='www.ensembl.org')
t2g_hs <- biomaRt::getBM(attributes = c('ensembl_transcript_id', 'ensembl_gene_id', 'external_gene_name', 'refseq_mrna'), mart = mart)
tx2gene <- t2g_hs[,c(1,2)]
tx2gene[nrow(tx2gene)+1,]=c('rRNA_45S_NR145819','rRNA_45S_NR145819')


ens2gene <- t2g_hs[,c(2,3)]
colnames(ens2gene)[2] <- 'Gene'
ens2gene <- unique(ens2gene)

# import transcript level counts
txi.salmon.t<-tximport(files, type="salmon", txOut=TRUE)
txi.salmon.g<-tximport(files=files, type="salmon", tx2gene= tx2gene, ignoreTxVersion = TRUE, countsFromAbundance = 'lengthScaledTPM' )
### this code works but if I remove the ignoreTxVersion I get an error, this may have to do with how I am generating my tx2gene file

# Extract counts only
counts <- txi.salmon.g$counts %>%
  as.data.frame()
#Extract TPM
tpms <- data.frame(txi.salmon.g$abundance)

##for clients the counts and tpm files should be written out

metadata <- read_excel("~/Desktop/Jordan files/counts.rRNA/meta.xlsx")
sums<-colSums(txi.salmon.g$counts)
sums<-as.data.frame(sums)
counts.rRNA<-as.data.frame(txi.salmon.g$counts)
rna.counts<-as.data.frame(t(slice_tail(counts.rRNA)))
rna.counts <-merge(rna.counts,sums,by='row.names',all=TRUE)
rna.counts<-mutate(rna.counts, "percentrRNA"= (rRNA_45S_NR145819 /sums)*100 )
names(rna.counts)[1] <- 'SampleName'
rna.counts<-dplyr::select(rna.counts, c("SampleName", "percentrRNA"))
metadata<-dplyr::full_join(metadata,rna.counts, by="SampleName")
metadata$percentrRNA <- round(metadata$percentrRNA ,digit=2)
kbl(metadata, booktabs=T, caption="Metadata Table") %>%
  kable_styling(latex_options=c("striped", "scale_down"))
```
There was little rRNA contamination in these samples.  This is expected as these are from polyA selected libraries.  


```{r, echo=FALSE}
expressed_genes<-tpms %>% filter_all(all_vars(. > 5))

```
There are 5001 genes with more than 5 counts in all samples

## Sample Heatmap and Correlation matrix
```{r All samples Heatmap, echo=FALSE, fig.width=7.5, fig.height=7, fig.align='center'}
## what we see here shows that patients cluster more closely than the cell types and this will need to be accounted for in the DGE
pheatmap(expressed_genes, scale="row")
cols.cor<-cor(expressed_genes, method="spearman")

pheatmap(cols.cor, method="spearman")
```
We see higher corrlation between samples than between treatments.  This is expected for patient data.  

## PCA plot



```{r pca plot, message=FALSE, echo=FALSE}
exp.pca <- prcomp(t(log2(expressed_genes)))
exp.pca.summary <- summary(exp.pca)$importance
pc1var = round(exp.pca.summary[3,1] * 100, 1)
pc2var = round(exp.pca.summary[3,2] * 100 - pc1var, 1)
exp.pca.pc <- data.frame(exp.pca$x, sample = colnames(expressed_genes))

pc_loadings<-exp.pca$rotation %>%
  as_tibble(rownames="ensembl_gene_id")
top_loading_pc1<-pc_loadings %>%
  dplyr::select(ensembl_gene_id, PC1) %>%
  arrange(desc(abs(PC1)))%>%
  inner_join(ens2gene, by="ensembl_gene_id") %>%
  head(n=10)

  top_loading_pc2<-pc_loadings %>%
  dplyr::select(ensembl_gene_id, PC2) %>%
  arrange(desc(abs(PC2)))%>%
  inner_join(ens2gene, by="ensembl_gene_id") %>%
  head(n=10)

```

### PC1 vs PC2
```{r, All samples PCA 1 vs 2, echo = FALSE, warning=FALSE,  fig.width= 6, fig.height=4.5, fig.align='center'}

colors <- c(viridis[6], cividis[9], magma[6])
ggplot(exp.pca.pc, aes(x = PC1, y = PC2, color=metadata$cellType, shape=metadata$patient)) +
  geom_point(size = 5)+
  scale_fill_manual(values = alpha(colors, 1) ) + 
  xlab(paste('PC1,', pc1var, '% explained var.')) +
  ylab(paste('PC2,', pc2var, '% explained var.')) +
  guides(fill=guide_legend(override.aes = list(color=colors))) +
  geom_text_repel(aes(label=metadata$SampleName), size=3, max.overlaps = Inf)+
  ggtitle("Expressed Genes in All Samples > 5 TPM") 
```
We see a stronger correlation between patient than between sample type.  PC1 is seperating the patients and the major loadings of this component are IFI30, HBB, MGST1, AHSP, and CREM.  PC2 is separating based on treatment and the major loadings of this component are FOS, IGHM, PUF60, SLC25A6, and ICAM.  

##  Run Differential Expression testing using DESeq2 and Calculate Gene Set Enrichment
### Compare 123pos vs 123neg, 123neg vs bulk, and 123pos vs bulk 
#### sig = padj <0.01 and abs(l2fc) >0.5
#### 

```{r DGE, echo=FALSE, message=FALSE, warning=FALSE, results=FALSE}
coldata<-metadata
coldata<-column_to_rownames(coldata, 'SampleName')
## need to confirm that all names are in the same order
all(rownames(coldata) %in% colnames(txi.salmon.g$counts))
all(rownames(coldata) == colnames(txi.salmon.g$counts))
coldata<-coldata[colnames(txi.salmon.g$counts),]
all(rownames(coldata) == colnames(txi.salmon.g$counts))

dds <- DESeqDataSetFromTximport(txi.salmon.g,
                              colData = coldata,
                              design = ~ patient + cellType)

dds<-estimateSizeFactors(dds)
sf<-as.data.frame(dds$sizeFactor)
sf<-rownames_to_column(sf, var="SampleName")
metadata<-inner_join(metadata, sf, by="SampleName")
names(metadata)[5] <- "sizeFactor"
metadata$sizeFactor <- round(metadata$sizeFactor ,digit=2)
### unfiltered data
dds.unfiltered <- DESeq(dds)
res.unfiltered <- results( dds.unfiltered)

##filtered data changed this from dds[rowMins(counts(dds))>5,] to a less stringent filtering
keep<-rowSums(counts(dds))>=10
#dds.filtered<-dds[rowMins(counts(dds))>5,]
dds.filtered<-dds[keep,]
dds.filtered<-DESeq(dds.filtered)
res.filtered<-results(dds.filtered)

###outcome the first filtering cut from 60000 genese to 14000 genes with the new filter 35000genes are left which is about 1/2
### need to subset to do pairwise comparison unclear if this keeps the patient design aspect


#res_123vs123HHT_unfiltered<-results( dds.unfiltered, contrast=c("cellType", "123pos", "123pos_HHT"))

res_123vs123HHT_filtered<-results(dds.filtered, contrast=c("cellType", "123pos", "123pos_HHT"))


## look at the numbers of genes meeting threshold the log fold change call is not changing things
#sum( res_123vs123HHT_unfiltered$pvalue < 0.01 & abs(res_123vs123HHT_unfiltered$log2FoldChange) >= 0.5, na.rm=TRUE )


sum(res_123vs123HHT_filtered$pvalue < 0.01 & abs(res_123vs123HHT_filtered$log2FoldChange) >= 0.5, na.rm=TRUE )


#sum(res_123vs123HHT_unfiltered$padj < 0.01, na.rm=TRUE)

sum(res_123vs123HHT_filtered$padj < 0.01, na.rm=TRUE)

#sum(res_123vs123HHT_unfiltered$padj < 0.05, na.rm=TRUE)
sum(res_123vs123HHT_filtered$padj < 0.05, na.rm=TRUE)
ens2gene <- t2g_hs[,c(2,3)]
ens2gene <- unique(ens2gene)
DEG_filtered<-as.data.frame(res_123vs123HHT_filtered)
DGE_filter<-filter(DEG_filtered, padj<.01)
DGE_filter<-rownames_to_column(DGE_filter, var="ensembl_gene_id")
DGE_filter<-right_join(ens2gene,DGE_filter, by="ensembl_gene_id") %>% dplyr::select(external_gene_name, padj, pvalue,log2FoldChange)
DGE_filter<-arrange(DGE_filter, padj) 
DGE_filter$padj<-formatC(DGE_filter$padj, format = "e", digits = 2)
DGE_filter$pvalue<-formatC(DGE_filter$pvalue, format = "e", digits = 2)
DGE_filter$log2FoldChange<-round(DGE_filter$log2FoldChange, digits = 2)

#DGE_filter$padj<-format(DGE_filter$padj, digits=3)
#DGE_filter$pvalue<-format(DGE_filter$pvalue, digits=3)
#DGE_filter$log2FoldChange<-format(DGE_filter$log2FoldChange, digits=3)

plotDispEsts(dds.filtered)
```

```{r, echo=FALSE}
kbl(DGE_filter, booktabs = T, longtable = TRUE, caption = "Genes with padj value of <.01") %>%
  kable_styling(latex_options = c("hold_position", "repeat_header"))%>%
  scroll_box(width = "600px", height = "500px")
```

Table showing the 132 genes with significant BH adjusted p- value of <.01.  

### Top 20 DEG plots
```{r, echo=FALSE, warning=FALSE, message=FALSE}
top_20_genes<-(DGE_filter[1:20, 1])
normalized_counts <- counts(dds.filtered, normalized=T)
normalized_counts<-as.data.frame(normalized_counts)
normalized_counts<-rownames_to_column(normalized_counts, var="ensembl_gene_id")
normalized_counts<-inner_join(normalized_counts, ens2gene, by="ensembl_gene_id")

normalized_counts<-dplyr::select(normalized_counts, external_gene_name, HTB336_CD123_pos, HTB336_CD123_pos_HHT, HTB61_CD123, HTB61_CD132_HHT)
top20_norm_counts <- filter(normalized_counts, external_gene_name %in% top_20_genes)
melted_top20_norm_counts <- data.frame(melt(top20_norm_counts))
colnames(melted_top20_norm_counts)<-c("gene", "SampleName", "normalized_counts")
melted_top20_norm_counts<-full_join(melted_top20_norm_counts,metadata, by="SampleName")
ggplot(melted_top20_norm_counts) +
        geom_point(aes(x = gene, y = normalized_counts, color = cellType)) +
        scale_y_log10() +
        xlab("Genes") +
        ylab("Normalized Counts") +
        ggtitle("Top 20 Significant DE Genes") +
        theme_bw() +
	theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
	theme(plot.title=element_text(hjust=0.5))
```

### Volcano Plot
```{r, echo=FALSE, warning=FALSE, fig.width= 10, fig.height=6}

colors <- c("black", viridis[5])
#df<-data.frame(res_123vs123HHT_unfiltered)
#df<-mutate(df, DiffExp=pvalue<=.05 & abs(log2FoldChange) >=.5)
#ggplot(df, aes(x=log2FoldChange, y=-log10(padj), color=DiffExp)) + 
#  geom_point() +
#  theme_light() +
#  scale_colour_manual(values = colors) +
#  ggtitle("123pos vs 123posHHT unfiltered")

colors <- c("black", viridis[5])
df<-data.frame(res_123vs123HHT_filtered)
df<-mutate(df, DiffExp=pvalue<=.05 & abs(log2FoldChange) >=.5)
ggplot(df, aes(x=log2FoldChange, y=-log10(padj), color=DiffExp)) + 
  geom_point() +
  theme_light() +
  scale_colour_manual(values = colors) +
  ggtitle("123pos vs 123posHHT filtered")

```
###MA plots
```{r, echo=FALSE, fig.width= 6, fig.height=4.5}

#DESeq2::plotMA(res_123vs123HHT_unfiltered, main="123pos vs 123posHHT unfiltered", 
#      ylim=c(-5,5))

DESeq2::plotMA(res_123vs123HHT_filtered,  main="123pos vs 123posHHT filtered", 
       ylim=c(-5,5))

```

```{r pca plot 2, message=FALSE, echo=FALSE, warning=FALSE}
pca_data<-vst(dds.filtered, blind=T)
ntop=500
rv <- rowVars(assay(pca_data))
select <- order(rv, decreasing=TRUE)[seq_len(min(ntop, length(rv)))]
pca <- prcomp(t(assay(pca_data)[select,]))
percentVar2 <- data.frame(percentVar=pca$sdev^2/sum(pca$sdev^2))%>%
  mutate(pc=1:n())%>%dplyr::select(pc, percentVar)
pca_df<-pca$x%>%data.frame()%>%mutate(type=metadata$cellType,percent=metadata$percentrRNA)

alt_col_values=c("#88CCEE", "#CC6677", "#DDCC77")

pc_loadings<-pca$rotation %>%
  as_tibble(rownames="ensembl_gene_id")
top_loading_pc1<-as.data.frame(pc_loadings) %>%
  dplyr::select(ensembl_gene_id, PC1) %>%
  arrange(desc(abs(PC1)))%>%
  inner_join(ens2gene, by="ensembl_gene_id")%>%
  head(n=10)

top_loading_pc2<-pc_loadings %>%
  dplyr::select(ensembl_gene_id, PC2) %>%
  arrange(desc(abs(PC2)))%>%
  inner_join(ens2gene, by="ensembl_gene_id") %>%
  head(n=10)


  
  
ggplot(pca_df, aes(PC1, PC2, color=type, shape=metadata$patient))+geom_point(size=3)+
  xlab(paste0("PC1: ",round(percentVar2$percentVar[1] * 100), "% variance")) + 
  ylab(paste0("PC2: ", round(percentVar2$percentVar[2] * 100), "% variance"))+
  scale_color_manual(values=alt_col_values)+labs(color="")+theme_bw()+
  ggtitle("PCA of normalized data")
```

This PCA plot shows the first 2 principle components following normalization with DEseq2.  Again we see PC1 separating by the patient with the top gene loadings of KYNU, MTC01P12, SLAMF7, CCL7, and PPBP.  The PC2 is separating on the treatment and the top gene loadings are LRPAP, FOS, TM4SF1, HBA1/2, and WASH5P.  

```{r gsea, echo=FALSE, warning=FALSE}
##pull gene names for gsea analysis
ens2gene <- t2g_hs[,c(2,3)]
colnames(ens2gene)[2] <- 'Gene'
ens2gene <- unique(ens2gene)

#loading hallmark and KEGG pathways

pathways.hallmark <- gmtPathways("~/Desktop/Jordan files/h.all.v7.4.symbols.gmt")
pathways.kegg<-gmtPathways("~/Desktop/Jordan files/c2.cp.kegg.v7.4.symbols.gmt")

##generating tidy data and adding it to the results file
res_123vs123HHT_filtered_gsea<-results(dds.filtered, contrast=c("cellType", "123pos", "123pos_HHT"),tidy=TRUE)
colnames(res_123vs123HHT_filtered_gsea)[1]<-"ensembl_gene_id"
res_123vs123HHT_filtered_gsea<-inner_join(res_123vs123HHT_filtered_gsea, ens2gene, by="ensembl_gene_id")
res_123vs123HHT_filtered_gsea_sum<-res_123vs123HHT_filtered_gsea%>%
  dplyr::select(Gene, stat) %>%
  na.omit() %>%
  distinct() %>%
  group_by(Gene) %>%
  summarise(stat=mean(stat))

rank_posHHT<-deframe(res_123vs123HHT_filtered_gsea_sum)
fgsea_posHHT_hallmark<- fgsea(pathways=pathways.hallmark, stats=rank_posHHT)
fgsea_posHHT_KEGG<-fgsea(pathways=pathways.kegg, stats=rank_posHHT)

fgseaRes_posHHT_tidy <-fgsea_posHHT_hallmark%>%
  as_tibble() %>%
  arrange(desc(NES))

fgseaRes_posHHT_kegg_tidy <-fgsea_posHHT_KEGG%>%
  as_tibble() %>%
  arrange(desc(NES))
```

## Displaying a table of ordered pathways

```{r, echo=FALSE, warning=FALSE}
fgseaRes_posHHT_tidy%>%
  dplyr::select(-leadingEdge, -ES) %>%
  arrange(padj) %>%
  DT::datatable(caption = 'Table 1: Hallmark genes with 123pos vs 123posHHT.')
```

```{r, echo=FALSE, warning=FALSE}
fgseaRes_posHHT_kegg_tidy%>%
  dplyr::select(-leadingEdge, -ES) %>%
  arrange(padj) %>%
  DT::datatable(caption = 'Table 1: KEGG pathway genes with 123pos vs 123posHHT.')
```
# plot the waterfall results

```{r, echo=FALSE, warning=FALSE, fig.width= 6, fig.height=8}
ggplot(fgseaRes_posHHT_tidy, aes(reorder(pathway, NES), NES)) +
  geom_col(aes(fill=padj<0.01)) +
  coord_flip() +
  labs(x="Pathway", y="Normalized Enrichment Score",
       title="Hallmark pathways NES from GSEA 123pos vs 123posHHT") + 
  theme_minimal()
```

```{r, echo=FALSE, warning=FALSE, fig.width= 8, fig.height=16}
ggplot(fgseaRes_posHHT_kegg_tidy, aes(reorder(pathway, NES), NES)) +
  geom_col(aes(fill=padj<0.01)) +
  coord_flip() +
  labs(x="Pathway", y="Normalized Enrichment Score",
       title="Kegg pathways NES from GSEA 123pos vs 123posHHT") + 
  theme_minimal()
```

#### __All Pathways In Moustache Plot__
```{r, echo=FALSE, warning=FALSE, include = FALSE}
# First calculate the q values
pvalues_hallmark <- fgseaRes_posHHT_tidy$pval

#fgseaRank <- fgseaResTidy[order(fgseaResTidy$pval),]
#qobj.df <- cbind.data.frame(qobj$pvalues, qobj$qvalues, qobj$lfdr)
#qobj.dfRank <- qobj.df[order(qobj.df$`qobj$pvalues`),]
  
toplotMoustache <- cbind.data.frame(fgseaRes_posHHT_tidy$pathway, fgseaRes_posHHT_tidy$NES, fgseaRes_posHHT_tidy$padj, fgseaRes_posHHT_tidy$pval)
colnames(toplotMoustache) <- c("pathway", "NES", "padj", "pval")

```

```{r, echo=FALSE, warning=FALSE}
ggplot(toplotMoustache, aes(x = NES, y = padj)) +
  geom_point() + 
  theme_minimal() +
  xlab('NES') + 
  ylab('BH adjusted p-value') +
  ggtitle("Hallmark pathways from GSEA") +
  geom_text_repel(aes(label=ifelse(padj<0.01,as.character(pathway),"")),hjust=-.5,vjust=-.5, size=1.5, max.overlaps = Inf) +
  labs(title = "Moustache plot of Hallmark pathways")+
  coord_cartesian(xlim = c(-4, 3), ylim = c(-0.1, 1),clip="off")
```

##repeating with KEGG pathways
```{r, echo = FALSE, warning=FALSE}
# First calculate the q values
pvalues_kegg <- fgseaRes_posHHT_kegg_tidy$pval

#fgseaRank <- fgseaResTidy[order(fgseaResTidy$pval),]
#qobj.df <- cbind.data.frame(qobj$pvalues, qobj$qvalues, qobj$lfdr)
#qobj.dfRank <- qobj.df[order(qobj.df$`qobj$pvalues`),]
  
toplotMoustache_kegg <- cbind.data.frame(fgseaRes_posHHT_kegg_tidy$pathway, fgseaRes_posHHT_kegg_tidy$NES, fgseaRes_posHHT_kegg_tidy$padj, fgseaRes_posHHT_kegg_tidy$pval)
colnames(toplotMoustache_kegg) <- c("pathway", "NES", "padj", "pval")

```

```{r, echo=FALSE, warning=FALSE}
ggplot(toplotMoustache_kegg, aes(x = NES, y = padj)) +
  geom_point() + 
  theme_minimal() +
  xlab('NES') + 
  ylab('BH adjusted p-value') +
  ggtitle("Kegg pathways from GSEA") +
  geom_text_repel(aes(label=ifelse(padj<0.05,as.character(pathway),"")),hjust=.5,vjust=0, size=1.5, max.overlaps = Inf )+
  coord_cartesian(xlim = c(-4, 4), ylim = c(-0.1, 1))
```

```{r, include=T, message=F, warning=FALSE, echo=FALSE}
topPathwaysUp <- fgsea_posHHT_hallmark[ES > 0][head(order(pval), n=10), pathway]
topPathwaysDown <- fgsea_posHHT_hallmark[ES < 0][head(order(pval), n=10), pathway]
topPathways <- c(topPathwaysUp, rev(topPathwaysDown))
```

```{r, include=T, message=F, warning=FALSE, echo=FALSE}
topPathwaysUp_kegg <- fgsea_posHHT_KEGG[ES > 0][head(order(pval), n=10), pathway]
topPathwaysDown_kegg <- fgsea_posHHT_KEGG[ES < 0][head(order(pval), n=10), pathway]
topPathways_kegg <- c(topPathwaysUp_kegg, rev(topPathwaysDown_kegg))
```

#### __Top Up and Down Ranked Pathways hallmark__
```{r, echo=FALSE, warning=FALSE, fig.width= 9, fig.height=5, fig.align='center'}
plotGseaTable(pathways.hallmark[topPathways], rank_posHHT, fgsea_posHHT_hallmark, 
              gseaParam=0.5, colwidths = c(7, 3, 0.8, 1.0, 1.0))
```

#### __Top Up and Down Ranked Pathways kegg __
```{r, echo=FALSE, warning=FALSE, fig.width= 9, fig.height=5, fig.align='center'}
plotGseaTable(pathways.kegg[topPathways_kegg], rank_posHHT, fgsea_posHHT_KEGG, 
              gseaParam=0.5, colwidths = c(7, 3, 0.8, 1.0, 1.0))
```

#### __Top ranked pathways hallmark__
**I can do this for any of interest, just let me know**
```{r, echo=FALSE, warning=FALSE, fig.width= 6, fig.height=4.5, fig.align='center'}
top.path <- as.character(fgseaRes_posHHT_tidy[1,1])
plotEnrichment(pathways.hallmark[[top.path]],
               rank_posHHT) + labs(title=top.path)
```

```{r, echo=FALSE, warning=FALSE, fig.width= 6, fig.height=4.5, fig.align='center'}
top.path <- as.character(fgseaRes_posHHT_tidy[50,1])
plotEnrichment(pathways.hallmark[[top.path]],
               rank_posHHT) + labs(title=top.path)
```

```{r, echo=FALSE, warning=FALSE, fig.width= 6, fig.height=4.5, fig.align='center'}
top.path <- as.character(fgseaRes_posHHT_tidy[49,1])
plotEnrichment(pathways.hallmark[[top.path]],
               rank_posHHT) + labs(title=top.path)
```

#### __Top ranked pathways kegg__
**I can do this for any of interest, just let me know**
```{r, echo=FALSE, warning=FALSE, fig.width= 6, fig.height=4.5, fig.align='center'}
top.path <- as.character(fgseaRes_posHHT_kegg_tidy[1,1])
plotEnrichment(pathways.kegg[[top.path]],
               rank_posHHT) + labs(title=top.path)
```
```{r, echo=FALSE, warning=FALSE, fig.width= 6, fig.height=4.5, fig.align='center'}
top.path <- as.character(fgseaRes_posHHT_kegg_tidy[2,1])
plotEnrichment(pathways.kegg[[top.path]],
               rank_posHHT) + labs(title=top.path)
```

```{r, echo=FALSE, warning=FALSE, fig.width= 6, fig.height=4.5, fig.align='center'}
top.path <- as.character(fgseaRes_posHHT_kegg_tidy[185,1])
plotEnrichment(pathways.kegg[[top.path]],
               rank_posHHT) + labs(title=top.path)
```


```{r, echo=FALSE, warning=FALSE, fig.width= 6, fig.height=4.5, fig.align='center'}
top.path <- as.character(fgseaRes_posHHT_kegg_tidy[186,1])
plotEnrichment(pathways.kegg[[top.path]],
               rank_posHHT) + labs(title=top.path)
```
## Session Information

```{r}
sessionInfo()
```